default.nix

```diff
@@ -0,0 +1,49 @@
+# Default import pinned pkgs
+{ pkgsSrc ? (import ./nix/pkgs.nix {}).pkgsSrc
+, pkgs ? (import ./nix/pkgs.nix { inherit pkgsSrc; }).pkgs
+, doCheck ? true
+}: with pkgs;
+
+let
+  # Get contract dependencies from lock file
+  inherit (callPackage ./nix/dapp.nix {}) specs packageSpecs;
+  inherit (specs.this) deps;
```

Comment:
What's `this`?

Reply:
`this` refers to the `dapp2nix` spec (the lock data for one repo) of the current repo. I.e. `testchain-dss-deployment-scripts`.

Reply:
Where is it defined? Is this some kind of reflection in nix or does dapp2nix define a `this` key?

Reply:
Maybe a poorly chosen name considering JavaScripts `this`, but this is just a key that I chose when generating the lock file with `dapp2nix` https://github.com/icetan/dapp2nix/blob/master/dapp2nix#L61